model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/userSession/HotRodUserSessionTransaction.java

```diff
@@ -65,6 +66,7 @@ public void commit() {
     }
 
     private MapAuthenticatedClientSessionEntity wrapClientSessionEntityToClientSessionAwareDelegate(MapAuthenticatedClientSessionEntity d) {
+        if (clientSessionTransaction.read(d.getId()) == null) return null;
```

Comment:
One more thing I forgot, could you please check whether it is possible to use exists/contains on cache instead of read?

Reply:
I tried the approach you proposed. One caveat is with `exists()` method in `CHM` storage where it can return `true` for expired entities. Is it OK or we should try to fix it somehow?

Reply:
We should not return true for entities that are expired. I think we can just provide a default implementation for `exists` equal to `read(id) != null`

Reply:
done